src/TurboRouter.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.10;
+
+import {TurboMaster} from "./TurboMaster.sol";
+import {TurboSafe} from "./TurboSafe.sol";
+import {RouterAuth} from "./authorities/RouterAuth.sol";
+import {ERC20} from "solmate-next/tokens/ERC20.sol";
+import {SafeTransferLib} from "solmate-next/utils/SafeTransferLib.sol";
+
+import {ERC4626} from "solmate-next/mixins/ERC4626.sol";
+import {Auth, Authority} from "solmate-next/auth/Auth.sol";
+
+/**
+ @title a router which can perform multiple Turbo actions between Master and the Safes
+ @notice routes custom users flows between actions on the master and safes.
+         Accomodates use cases where the Safe and actions on the created Safe happen in the same transaction.
+         Supported Flows:
+          * creating a safe
+          * Auth methods (set owner/authority)
+          * ERC-4626 methods (deposit/withdraw/mint/redeem)
+          * TurboSafe methods (boost/less/sweep/slurp)
+ */
+contract TurboRouter is RouterAuth {
+    using SafeTransferLib for ERC20;
+
+    TurboMaster public immutable master;
+
+    uint256 private constant _NOT_IN_MULTICALL = 0;
+    uint256 private constant _IN_MULTICALL = 1;
+
+    uint256 private _multicallStatus;
```

Comment:
all mutex lock variables should be packed into smaller data types to save on gas if you're going to use this solution

Reply:
yeah good point could pack the safe and status